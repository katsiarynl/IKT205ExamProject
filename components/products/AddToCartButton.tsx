import { View, Text, Pressable } from "react-native";
import React, { useContext } from "react";
import products from "../../styles/products";
import { RestaurantContext } from "../../context";
import { Product } from "../../types/productTypes";
import Toast from "react-native-toast-message";
export default function AddToCartButton(props: Product) {
  const { dispatch } = useContext(RestaurantContext);

  const new_product = {
    _id: props._id,
    name: props.name,
    description: props.description,
    price: props.price,
    restaurant: props.restaurant,
    image: props.image,
  };
  const fundisp = (dispatch, cartitem: Product) => {
    dispatch(
      (() => {
        return { type: "PLACE_ORDER", payload: cartitem };
      })()
    );
  };

  const alertAddedToCart = () => {
    fundisp(dispatch, new_product);

    Toast.show({
      type: "success",
      text1: `${new_product.name} Has been added to your cart.`,
      text2: `${new_product.price + "NOK"} View Your cart `,
      position: "top",
      autoHide: true,
      topOffset: 40,
      bottomOffset: 90,
    });
  };

  return (
    <View style={{ flex: 1 }}>
      <Pressable
        onPress={() => {
          alertAddedToCart();
        }}
        //https://www.youtube.com/watch?v=ooEFRONfq_s
        style={({ pressed }) => pressed && { opacity: 0.4 }}
      >
        <View style={products.addTocartButton}>
          <Text style={{ fontSize: 18, fontWeight: "bold" }}>Add To Cart</Text>
        </View>
      </Pressable>
    </View>
  );
}
