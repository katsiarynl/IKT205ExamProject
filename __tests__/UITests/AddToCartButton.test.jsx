import React from "react";
import { create } from "react-test-renderer";
import AddToCartButton from "../../components/products/AddToCartButton";
const tree = create(<AddToCartButton />);

test("Snapshot for AddToCartButton", () => {
  expect(tree).toMatchSnapshot();
});
