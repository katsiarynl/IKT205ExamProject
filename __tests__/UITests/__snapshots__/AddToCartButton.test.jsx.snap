// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for AddToCartButton 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={false}
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "orange",
          "borderRadius": 15,
          "height": 80.03999999999999,
          "justifyContent": "center",
          "width": 210.00000000000003,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
      >
        Add To Cart
      </Text>
    </View>
  </View>
</View>
`;
